src/Polynomials.jl

```diff
@@ -372,7 +372,7 @@ function polyval{T,S}(p::Poly{T}, x::S)
     if lenp == 0
         return zero(R) * x
     else
-        y = convert(R, p[end]) + zero(T)*x
+        y = convert(R, p[end])
```

Comment:
actually, I do not see any benefit in using `zero(T)*zero(x)`. since we have `convert(R, p[end])` where `R = promote_type(T,S)`, `zero(x)` will be automatically promoted to `R`. For normal numbers, I mean standard `Number`s, `zero(x)` is not needed at all. Then, if there is a custom type `Type <: Number`, it might make sense to use `zero(x)` and rely on the conversion rules among `Number`s.

What do you think?

Do you think we can push these issues (i.e., Horner's bug, `Base./` and perhaps `polyder`) this weekend?